.github/workflows/weekly_ci.yml

```diff
@@ -12,56 +12,66 @@ jobs:
   weekly_ci:
     runs-on: ubuntu-latest
     steps:
-      - name: Delete the previous branch if exists
-        uses: actions/github-script@v6
-        with:
-          github-token: ${{ secrets.GITHUB_TOKEN }}
-          script: |
-            const response = await github.rest.git.listMatchingRefs({
-              owner: context.repo.owner,
-              repo: context.repo.repo,
-              branch: "heads/${{ env.BRANCH_NAME }}",
-            });
-            for (ref in response.data){
-              if (response.data[ref].ref == "refs/heads/${{ env.BRANCH_NAME }}"){
-                console.log("deleting branch");
-                console.log(response.data[ref]);
-                await github.rest.git.deleteRef({
-                  owner: context.repo.owner,
-                  repo: context.repo.repo,
-                  ref: "heads/${{ env.BRANCH_NAME }}",
-                });
-              }
-            }
-
       - uses: actions/checkout@v3
         with:
           fetch-depth: 0
 
       - name: Create and push empty commit
         run: |
           set -euxo pipefail
+
+          get_main_branch() {
+              local br
+              br=$(git branch --list main)
+              if [ -n "$br" ]; then
+                  echo main
+                  return 0
+              fi
+              br=$(git branch --list master)
+              if [ -n "$br" ]; then
+                  echo master
+                  return 0
+              fi
+              echo UNKNOWN
+              return 1
+          }
+
           git config --global user.name "github-actions[bot]"
           git config --global user.email "41898282+github-actions[bot]@users.noreply.github.com"
-          git checkout -b weekly-ci
+          git checkout ${{ env.BRANCH_NAME }} || git checkout -b ${{ env.BRANCH_NAME }}
+          git rebase $(get_main_branch)
           git commit --allow-empty -m "${{ env.COMMIT_MESSAGE }}"
-          git push --set-upstream origin ${{ env.BRANCH_NAME }}
+          git push -f --set-upstream origin ${{ env.BRANCH_NAME }}
 
       - name: Create and comment pull request
         uses: actions/github-script@v6
         with:
-          github-token: ${{ secrets.GH_PUSH_TOKEN }}
+          github-token: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
I think the comment still has to be created by the user which has the token `GH_PUSH_TOKEN` - otherwise, the citest comment won't trigger ci

Reply:
Oh, sorry, this is leftover, will change it back. Thanks for noticing